### PR TITLE
Show when showOptionsWhenEmpty is true only if the input is focused

### PR DESCRIPTION
**Pull request by gregeinfrank**
There is an config option for `showOptionsWhenEmpty`, but the problem is there is no way to hide the options if you use it. What we really want is to show the options when you click into the input (on focus) if there is no input text.

### CODE REVIEW

**Comment by gregeinfrank** [comment]
@fmoo can you please take a look?

File: src/typeahead/index.js [hunk 1]
```diff
@@ -88,13 +88,16 @@ var Typeahead = React.createClass({
       selection: this.props.value,
 
       // Index of the selection
-      selectionIndex: null
+      selectionIndex: null,
+
+      isFocused: false,
```
Comment (fmoo) [inline src/typeahead/index.js]: Add a comment?

File: src/typeahead/index.js [hunk 2]
```diff
@@ -88,13 +88,16 @@ var Typeahead = React.createClass({
...
     };
   },
 
   _shouldSkipSearch: function(input) {
     var emptyValue = !input || input.trim().length == 0;
-    return !this.props.showOptionsWhenEmpty && emptyValue;
+    var isFocused = this.state && this.state.isFocused;
```
Comment (fmoo) [inline src/typeahead/index.js]: Why do you need the `this.state &&` part of this?

Comment (gregeinfrank) [inline src/typeahead/index.js]: It's kind of funky, but the reason is that `_shouldSkipSearch` is called from `getOptionsForValue`, which is called from `getInitialState`. Within `getInitialState`, `this.state` is not yet defined, so this has to be special cased.

Comment (fmoo) [inline src/typeahead/index.js]: > `_shouldSkipSearch` is called from `getOptionsForValue`, which is called from `getInitialState`

Oh ew, that's right.  Great catch!

**Pull request merged**